src/common/timer_manager.cc

```diff
@@ -0,0 +1,58 @@
+#include "timer_manager.h"
+
+#include "common/logging.h"
+#include "common/util.h"
+
+namespace raftcpp {
+namespace common {
+
+TimerManager::TimerManager() {
+    io_service_ = std::make_unique<asio::io_service>();
+    work_ = std::make_unique<asio::io_service::work>(*io_service_);
+}
+
+TimerManager::~TimerManager() {
+    for (auto &timer : timers_) {
+        timer->Stop();
+    }
+
+    io_service_->stop();
+    thread_->join();
+}
+
+void TimerManager::Run() {
+    // Note that the `Start()` should be invoked before `io_service->run()`.
+    thread_ = std::make_unique<std::thread>([this]() { io_service_->run(); });
+}
+
+void TimerManager::StartTimer(int id, uint64_t timeout_ms) {
+    RAFTCPP_CHECK(0 <= id < static_cast<int>(timers_.size()));
+    timers_[id]->Start(timeout_ms);
+}
+
+void TimerManager::ResetTimer(int id, uint64_t timeout_ms) {
+    RAFTCPP_CHECK(0 <= id < static_cast<int>(timers_.size()));
+    timers_[id]->Reset(timeout_ms);
+}
+
+void TimerManager::StopTimer(int id) {
+    RAFTCPP_CHECK(0 <= id < static_cast<int>(timers_.size()));
+    timers_[id]->Stop();
+}
+int TimerManager::RegisterTimer(const std::function<void(void)> &handler) {
+    if (handler == nullptr) {
+        return -1;
+    }
+
+    timers_.emplace_back(std::make_shared<common::RepeatedTimer>(
+        *io_service_, [handler](const asio::error_code &e) {
```

Comment:
C++14 supports std::move on lambda capture:
```suggestion
        *io_service_, [std::move(handler)](const asio::error_code &e) {
```

Reply:
const ref， no need std::move ?